Origin:  http://www.spinics.net/lists/linux-kbuild/msg05612.html

This patch fixes the problem that "conf --randconfig" does not handle choices
as expected. As of 3.1-rc9 this patch for rc4 ist still not available upstream.

--- scripts/kconfig/confdata.c	2013-01-13 08:53:12+01:00
+++ scripts/kconfig/confdata.c	2013-01-13 08:53:12+01:00
@@ -1077,6 +1077,9 @@
 		else {
 			sym->def[S_DEF_USER].tri = no;
 		}
+		sym->flags |= SYMBOL_DEF_USER;
+		/* clear VALID to get value calculated */
+		sym->flags &= ~(SYMBOL_VALID);
 	}
 	csym->flags |= SYMBOL_DEF_USER;
 	/* clear VALID to get value calculated */
